xcp_d/utils/plot.py

```diff
@@ -1006,3 +1006,127 @@ def _get_tr(img):
     except AttributeError:  # Error out if not in cifti
         return img.header.get_zooms()[-1]
     raise RuntimeError("Could not extract TR - unknown data structure type")
+
+
+def plot_alff_reho_volumetric(output_path, filename, bold_file):
+    """Plot ALFF/ReHo mosaic plot for Niftis."""
+    from bids.layout import parse_file_entities
+    from nilearn import plotting as plott
+    from templateflow.api import get as get_template
+
+    entities_to_use = ["cohort", "den", "res"]
+    space = parse_file_entities(bold_file)["space"]
+    file_entities = parse_file_entities(bold_file)
+    entities_to_use = {f: file_entities[f] for f in file_entities if f in entities_to_use}
+    template_file = get_template(template=space, **entities_to_use, suffix='T1w', desc=None)
+    if isinstance(template_file, list):
+        template_file = template_file[0]
+    template = str(template_file)
+    plott.plot_stat_map(filename,
+                        bg_img=template,
+                        display_mode='z',
+                        cut_coords=8,
+                        output_file=output_path)
+    return output_path
+
+
+def surf_data_from_cifti(data, axis, surf_name):
+    """From https://neurostars.org/t/separate-cifti-by-structure-in-python/17301/2.
+
+    https://nbviewer.org/github/neurohackademy/nh2020-curriculum/blob/master/we-nibabel-markiewicz/NiBabel.ipynb
+    """
+    assert isinstance(axis, nb.cifti2.BrainModelAxis)
+    for name, data_indices, model in axis.iter_structures():
+        # Iterates over volumetric and surface structures
+        if name == surf_name:  # Just looking for a surface
+            data = data.T[data_indices]
+            # Assume brainmodels axis is last, move it to front
+            vtx_indices = model.vertex
+            # Generally 1-N, except medial wall vertices
+            surf_data = np.zeros((vtx_indices.max() + 1,) + data.shape[1:], dtype=data.dtype)
+            surf_data[vtx_indices] = data
+            return surf_data
+
+    raise ValueError(f"No structure named {surf_name}")
+
+
+def plot_alff_reho_surface(output_path, filename, bold_file):
+    """Plot ReHo and ALFF for ciftis on surface."""
+    import matplotlib.pyplot as plt
+    import nibabel as nb
+    from bids.layout import parse_file_entities
+    from nilearn import plotting as plott
+    from templateflow.api import get as get_template
+
+    from xcp_d.utils.plot import surf_data_from_cifti
+
+    func = filename
+    density = parse_file_entities(bold_file).get("den", "32k")
+    rh = str(
+        get_template(template="fsLR", hemi="L", density=density,
+                     suffix="midthickness")
+    )
+    lh = str(
+        get_template(template="fsLR", hemi="R", density=density, suffix="midthickness")
+    )
```

Comment:
I'd grab the space from the file_entities like you do with the volumetric function, since we plan to support fsaverage in the future.